<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>453</width>
    <height>889</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Experiment settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout" rowstretch="0,0,0,0" columnstretch="0,1,0">
      <item row="2" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Geometry:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QToolButton" name="load_detector">
        <property name="toolTip">
         <string>select spline file graphically</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Detector:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QWidget" name="widget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="file_import">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Import from file...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="poni_save">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Export...</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QToolButton" name="show_geometry">
        <property name="toolTip">
         <string>select spline file graphically</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="GeometryLabel" name="geometry_label">
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="DetectorLabel" name="detector_label">
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="textInteractionFlags">
         <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="UnitLabel" name="wavelengthUnit">
        <property name="text">
         <string>keV</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Wavelength:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QuantityEdit" name="wavelengthEdit">
        <property name="toolTip">
         <string>File containing the image of the mask</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Preprocessing</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="4" column="0">
       <widget class="QCheckBox" name="do_polarization">
        <property name="toolTip">
         <string>Perform polarization corrections</string>
        </property>
        <property name="text">
         <string>Polarization factor:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="do_dummy">
        <property name="toolTip">
         <string>Perform dynamic masking</string>
        </property>
        <property name="statusTip">
         <string/>
        </property>
        <property name="text">
         <string>Dummy value:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="do_mask">
        <property name="toolTip">
         <string>Mask-out pixels</string>
        </property>
        <property name="text">
         <string>Mask file:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="val_dummy">
        <property name="toolTip">
         <string>Dummy value for dynamic masking</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1" colspan="3">
       <widget class="FileEdit" name="flat_field">
        <property name="toolTip">
         <string>File containing the flat-field image</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>±</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="LoadImageToolButton" name="file_dark_current">
        <property name="toolTip">
         <string>Select a dark current file</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="LoadImageToolButton" name="file_flat_field">
        <property name="toolTip">
         <string>Select a flat field file</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="do_flat">
        <property name="toolTip">
         <string>Perform flat field corrrections</string>
        </property>
        <property name="text">
         <string>Flat field:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="do_dark">
        <property name="toolTip">
         <string>Perform dark current subtraction</string>
        </property>
        <property name="text">
         <string>Dark current:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="LoadImageToolButton" name="file_mask_file">
        <property name="toolTip">
         <string>Select a mask file</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="3">
       <widget class="FileEdit" name="mask_file">
        <property name="toolTip">
         <string>File containing the image of the mask</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="3">
       <widget class="FileEdit" name="dark_current">
        <property name="toolTip">
         <string>File containing the dark image</string>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QLineEdit" name="delta_dummy">
        <property name="toolTip">
         <string>Precision for dummy value</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1" colspan="3">
       <widget class="QDoubleSpinBox" name="polarization_factor">
        <property name="toolTip">
         <string>Between 1 (Vertical) and -1 (Horizontal polarization). </string>
        </property>
        <property name="minimum">
         <double>-1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QCheckBox" name="do_solid_angle">
        <property name="toolTip">
         <string>Correct solid angle effects</string>
        </property>
        <property name="text">
         <string>Solid angle corrections</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="widget_2" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QGroupBox" name="groupBox_8">
        <property name="title">
         <string>Radial integration</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_8">
         <item row="2" column="0">
          <widget class="QLabel" name="label_21">
           <property name="text">
            <string>Number of points:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="nbpt_rad"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>Radial unit:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="UnitSelector" name="radial_unit">
           <property name="sizeAdjustPolicy">
            <enum>QComboBox::AdjustToMinimumContentsLength</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="do_2D">
        <property name="title">
         <string>2D integration</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_7">
         <item row="1" column="0">
          <widget class="QLabel" name="label_20">
           <property name="text">
            <string>Number of points:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>Azimuthal unit:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="nbpt_azim"/>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="azimuthal_unit">
           <property name="text">
            <string>χ (rad)</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="title">
      <string>Integration range</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="2" column="2">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QLineEdit" name="radial_range_max">
        <property name="toolTip">
         <string>upper radial bound</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="do_radial_range">
        <property name="toolTip">
         <string>Limit/Force the radial range</string>
        </property>
        <property name="text">
         <string>Radial range:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="radial_range_min">
        <property name="toolTip">
         <string>Lower radial bound</string>
        </property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="QLabel" name="_radialRangeUnit">
        <property name="text">
         <string>?</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QCheckBox" name="chi_discontinuity_at_0">
        <property name="toolTip">
         <string>if unset, the discontinuity is at 180°</string>
        </property>
        <property name="text">
         <string>χ discontinuity at 0</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="do_azimuthal_range">
        <property name="toolTip">
         <string>Limit/force the azimuthal range</string>
        </property>
        <property name="text">
         <string>Azimuthal range:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="azimuth_range_min">
        <property name="toolTip">
         <string>Lower azimuthal bound</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QLineEdit" name="azimuth_range_max">
        <property name="toolTip">
         <string>Upper azimuthal bound</string>
        </property>
       </widget>
      </item>
      <item row="3" column="4">
       <widget class="QLabel" name="label_24">
        <property name="text">
         <string>deg</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>Processing</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QCheckBox" name="do_poisson">
        <property name="text">
         <string>Propagation of error:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="3">
       <widget class="QComboBox" name="error_selection">
        <item>
         <property name="text">
          <string>Poisson law</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="do_normalization">
        <property name="toolTip">
         <string>Perform normalization of the result of the integration, by dividing by this factor</string>
        </property>
        <property name="text">
         <string>Normalization:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="normalization_factor">
        <property name="toolTip">
         <string>Normalization factor applied to each integration</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>×</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="MonitorNameEdit" name="monitor_name">
        <property name="toolTip">
         <string>Monitor name to reach normalization from the header of the input data</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Algorithm:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="QToolButton" name="method_config_button">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="opencl_title">
        <property name="text">
         <string>OpenCL device:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="4">
       <widget class="QToolButton" name="opencl_config_button">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1" colspan="3">
       <widget class="MethodLabel" name="methodLabel">
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="3" column="1" colspan="3">
       <widget class="OpenClDeviceLabel" name="opencl_label">
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DetectorLabel</class>
   <extends>QLabel</extends>
   <header>pyFAI.gui.widgets.DetectorLabel</header>
  </customwidget>
  <customwidget>
   <class>UnitSelector</class>
   <extends>QComboBox</extends>
   <header>pyFAI.gui.widgets.UnitSelector</header>
  </customwidget>
  <customwidget>
   <class>OpenClDeviceLabel</class>
   <extends>QLabel</extends>
   <header>pyFAI.gui.widgets.OpenClDeviceLabel</header>
  </customwidget>
  <customwidget>
   <class>QuantityEdit</class>
   <extends>QLineEdit</extends>
   <header>pyFAI.gui.widgets.QuantityEdit</header>
  </customwidget>
  <customwidget>
   <class>UnitLabel</class>
   <extends>QLabel</extends>
   <header>pyFAI.gui.widgets.UnitLabel</header>
  </customwidget>
  <customwidget>
   <class>GeometryLabel</class>
   <extends>QLabel</extends>
   <header>pyFAI.gui.widgets.GeometryLabel</header>
  </customwidget>
  <customwidget>
   <class>MethodLabel</class>
   <extends>QLabel</extends>
   <header>pyFAI.gui.widgets.MethodLabel</header>
  </customwidget>
  <customwidget>
   <class>LoadImageToolButton</class>
   <extends>QToolButton</extends>
   <header>pyFAI.gui.widgets.LoadImageToolButton</header>
  </customwidget>
  <customwidget>
   <class>MonitorNameEdit</class>
   <extends>QLineEdit</extends>
   <header>pyFAI.gui.widgets.MonitorNameEdit</header>
  </customwidget>
  <customwidget>
   <class>FileEdit</class>
   <extends>QLineEdit</extends>
   <header>pyFAI.gui.widgets.FileEdit</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
